configure.ac

```diff
@@ -3959,9 +3959,17 @@ then
 	esac
 	case $host in
 	*-*-mingw*)
-	DYNLOADFILE="dynload_win.o"
-	extra_machdep_objs="$extra_machdep_objs PC/dl_nt.o"
-	CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"$VERSION\"' -DPY3_DLLNAME='\"libpython3.dll\"'"
+    DYNLOADFILE="dynload_win.o"
+    extra_machdep_objs="$extra_machdep_objs PC/dl_nt.o"
+    CFLAGS_NODIST="$CFLAGS_NODIST -DPY3_DLLNAME='\"libpython3.dll\"'"
+    case $host in
+        i686*)
+                CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"${VERSION}-32\"'"
```

Comment:
what about 32-bit arm?

Reply:
Hmm, doesn't this cover that case? 
The main reason why this is needed is that both 32-bit user-installed packages and 64-bit user-installed packages are installed in a different locations. This was an issue in the upstream previously and only fixed in 3.10 (I guess?), see https://bugs.python.org/issue43195 and https://bugs.python.org/issue41627. I could see in https://github.com/python/cpython/blob/fc54e722a2e66971f1a8e16cff24c844bf9c5ac4/PCbuild/python.props#L230 that MSVC python uses specific suffixes for arm, should we change according to that? Also, maybe also cover the cases where ucrt vs msvcrt?

Reply:
I think he means "-32" should also be used for `armv7*` maybe?

Reply:
or,
> MSVC python uses specific suffixes for arm, should we change according to that? Also, maybe also cover the cases where ucrt vs msvcrt?

if the intent of the change is to prevent mixing of extensions of different arches.

Reply:
Ideally yes. But I don't think we should use sys.winver then and use some other internal source for the "plat" ID. I don't want sys.winver to give a different result with our Python.

And this could be done in a different PR. This PR brings it at least in line with upstream.

Reply:
Although the current sys.winver docs don't even specify the "-32" suffix... so maybe could change it...hm..

Reply:
According to the above-referenced commit the suffixes used by upstream in sys.winver are
```xml
    <!-- Suffix for versions/keys when building for particular platforms -->
    <PyArchExt Condition="'$(ArchName)' == 'win32'">-32</PyArchExt>
    <PyArchExt Condition="'$(ArchName)' == 'arm32'">-arm32</PyArchExt>
    <PyArchExt Condition="'$(ArchName)' == 'arm64'">-arm64</PyArchExt>
```

Reply:
thanks. So it's not documented anyway.

So if we attach $PYD_PLATFORM_TAG there we'd get `3.10-mingw_x86_64_clang`?

I guess we can try.

Reply:
> But I don't think we should use sys.winver then and use some other internal source for the "plat" ID. 

In which I guess we would end up copying the [code in sysconfig for finding the platform](https://github.com/msys2-contrib/cpython-mingw/blob/2ea5fa382996dc48069fb18fe55491fafd8cbeb8/Lib/sysconfig.py#L702-L744) to `site.py`, since importing sysconfig in site.py isn't viable https://bugs.python.org/issue29585. Just confirming do we really want the user install site-package to be `~/.local/lib/python3.10-mingw_x86_64_clang` for CLANG64 python?

Also, if we aren't gonna use `sys.winver` for finding the user-install site-package I think making it follow upstream seems like a nice idea. 

Also, @jeremyd2019 do you know what is the `$MINGW_CHOST` for arm32 and arm64?

Reply:
See https://github.com/msys2-contrib/cpython-mingw/blob/aa916ed5c97093b9f8a7f585b249f1ec06b24aa9/configure.ac#L5197-L5202